### PR TITLE
Added preliminary project documentation

### PR DESCRIPTION
**Pull request by ohessel**
I hope I pictured everything correctly in consolidating the existing documents and documentation

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -1,7 +1,100 @@
...
+The following components will set up our orchestration system:
+
+* Kubernetes API-Server (in future: Orchestrator)
+* Agent(s): one per VM
+* Operators: one per application/Big Data Product
+
+== Components and their functionality
+=== API-Server (Orchestrator)
+At the moment we use the standard Kubernetes API-Server and stay API-compatible with Kubernetes. In future we might implement our own Orchestrator component if needed.
+The API-Server acts as a centralized storage for configuration files. It has the function to distribute the updated files and configurations or inform the other components about the current changes.
```
Comment (lfrancke) [inline README.adoc:45]:
```suggestion
The API-Server acts as a centralized frontend for the clusters shared storage. It is the component through which all other components (i.e. Agents, Kubelet, Operators, ...) communicate.
```

It doesn't really deal in "files"

File: README.adoc [hunk 2]
```diff
@@ -1,7 +1,100 @@
...
+
+=== Operator(s)
+For each application/Big Data Product there is exactly one special Operator defined. They must be registered to the API-Server for getting notified about upcoming configuration changes.
```
Comment (lfrancke) [inline README.adoc:48]:
```suggestion
For each application/Big Data Product there is exactly one special Operator defined.
They watch for changes in their managed resources in the API-Server.
```

File: README.adoc [hunk 3]
```diff
@@ -1,7 +1,100 @@
...
+This Operator is the expert of the application logic, how to roll out and update the Big Data Product (e.g. Kafka with the {base-repo}/kafka-operator[kafka-operator] and {base-repo}/zookeeper-operator[zookeeper-operator]). It can be compared like the Operator knows everything about the construction manual.
```
Comment (lfrancke) [inline README.adoc:49]:
```suggestion
This Operator is the expert of the application logic, how to roll out and update the Big Data Product (e.g. Kafka with the {base-repo}/kafka-operator[kafka-operator] and {base-repo}/zookeeper-operator[zookeeper-operator]).
An Operator should implement and encompass all the knowledge a human operator would need when operating the service in question.
```

File: README.adoc [hunk 4]
```diff
@@ -1,7 +1,100 @@
...
+To ease up operator development we created an {base-repo}/operator-rs[operator-framework] to base new operators on.
+
+
+=== Agent(s)
+Despite the Operator(s) the {base-repo}/agent[Agent(s)] are clones. But each Agent is running on its own VM. They must also be registered to the central API-Server, to get informed about updates.
```
Comment (lfrancke) [inline README.adoc:54]: I don't understand the first sentence. Could you explain what you mean?

File: README.adoc [hunk 5]
```diff
@@ -1,7 +1,100 @@
...
+The Agent(s) do not know anything about the Big Data product. They execute only their work instructions, which they receive from the Operator(s).
+
+
+== Communication and interaction
+image::interaction-diagram.svg[]
+
+
+=== 1. Client -> API-Server (Orchestrator)
+The client creates a YAML file for the API-Server, which defines the configuration for the Big Data Product (e.g. Kafka). There will be more possibilities to upload the file to the API-Server in future, because it is compatible to K8s and therefore every Kubernetes Tool can be used. Usually it is the command line tool ‘kubectl’
```
Comment (lfrancke) [inline README.adoc:63]:
```suggestion
The client creates a YAML file for the API-Server, which defines the configuration for the Big Data Product (e.g. Kafka). The client then uses the usual Kubernetes tools (e.g. `kubectl` or any other Kubernetes client).
This means that the whole infrastructure will be described in code (Infrastructure-as-Code) and can be version controlled.
```

File: README.adoc [hunk 6]
```diff
@@ -1,7 +1,100 @@
...
+Finally the Agent(s) will provision the Big Data Product (e.g. setting up a Kafka) onto the node of its VM.
+
+
+== Technical base
+=== API-Server (Orchestrator)
+The https://kubernetes.io/docs/reference/command-line-tools-reference/kube-apiserver/[Kubernetes API-Server] validates and configures data for the api objects which include pods, services, replicationcontrollers, and others. The API-Server services **REST** operations and provides the frontend to the cluster's shared state through which all other components interact.
+
+
+=== Operator(s)
+The Operator(s), which are can run on the API-Server or on their own server, can be written in **Java** or in every other language you want. In the current implementation **Rust** is used for the Zookeeper and Kafka Operator and **Java** for the Spark Operator.
```
Comment (lfrancke) [inline README.adoc:84]:
```suggestion
The Operator(s), which can run on the same server as the API-Server or on their own server, can be written in any programming language. We are using **Rust** to implement our Operators.
```

File: README.adoc [hunk 7]
```diff
@@ -1,7 +1,100 @@
...
+
+=== Agent(s)
+We have chosen **Rust** as a programming language for the implementation of the Agent. This tool will manage the installation on the VM.
+Rust is a Programming language, which is focused on safety. It provides memory safety without a garbage collector, however the learning curve is steep.
```
Comment (lfrancke) [inline README.adoc:88]:
```suggestion
Rust is a Programming language, which is focused on safety. It provides memory safety without a garbage collector.
```

No need to discourage anyone :)

File: README.adoc [hunk 8]
```diff
@@ -1,7 +1,100 @@
...
+The API-Server receives the YAML file from the Client, identifies the appropriate Operator(s) (e.g. Kafka Operator) and notifies about the file changes.
+
+=== 3. Operator(s) -> API-Server (Orchestrator)
+The appropriate Operator(s) (e.g. Kafka Operator) will translate the abstract description into detailed actions for the Agent(s).
+
+=== 4. Agent(s) -> API-Server (Orchestrator)
+In the meantime each Agent, running on its own VM, is listening for updates, which belongs to its VM. If there is an update for the appropriate VM, then the Agent of this VM is going to obtain the configuration and act accordingly. (e.g. download an artifact, update a configuration or start a process locally.
+
+=== 5. Agent(s) -> Big Data Product (Kafka)
+Finally the Agent(s) will provision the Big Data Product (e.g. setting up a Kafka) onto the node of its VM.
```
Comment (lfrancke) [inline README.adoc:75]:
```suggestion
Finally, the Agent(s) will provision the Big Data Product (e.g. setting up a Kafka) onto the node of its VM.
```

File: README.adoc [hunk 9]
```diff
@@ -1,7 +1,100 @@
...
+=== Operator(s)
+The Operator(s), which are can run on the API-Server or on their own server, can be written in **Java** or in every other language you want. In the current implementation **Rust** is used for the Zookeeper and Kafka Operator and **Java** for the Spark Operator.
+
+=== Agent(s)
+We have chosen **Rust** as a programming language for the implementation of the Agent. This tool will manage the installation on the VM.
+Rust is a Programming language, which is focused on safety. It provides memory safety without a garbage collector, however the learning curve is steep.
+
+**Decision Drivers:**
+  - The ability to deploy the agent as one binary with no external dependencies
+  - Availability of well supported libraries for necessary operations
```
Comment (lfrancke) [inline README.adoc:92]:
```suggestion
  - Availability of well-supported libraries for necessary operations
```

File: README.adoc [hunk 10]
```diff
@@ -1,7 +1,100 @@
...
+	- File IO
+	- Network IO
+	- systemd
+	- IDE support
+	- Debugging options
```
Comment (lfrancke) [inline README.adoc:97]: Could you replace tabs with spaces here?

**Review by lfrancke — CHANGES_REQUESTED** [review]

File: README.adoc [hunk 11]
```diff
@@ -1,7 +1,100 @@
...
+{empty}
+
+
 == Introduction
 
-tbd
+With stackable-tech we aim to create an orchestration system that allows end users to deploy, scale and manage Big Data infrastructure on any environment, no matter if it's VM or bare metal both in cloud and on-premise.
+
+== Goal of the project
+The goal is to be able to declare full Data Platforms using this concept and go beyond pure low-level applications. This means that we want to be able to deploy a Kafka cluster for example. In future we would also like to be able to declaratively roll out data, provide proper security configuration out of the box, Kafka topics, HDFS directories, Replication policies, maybe even Spark jobs.
```
[the remark this answers is not in the document]
Comment (ohessel) [reply README.adoc:21]: I think it the better description for "Kafka Cluster" would be.

"a Kafka cluster together with Zookeeper, an authorization framework and a Spark cluster to consume a data stream"

File: README.adoc (hunk 3 above)
Comment (ohessel) [reply README.adoc:49]: Coole Formulierung!

File: README.adoc (hunk 4 above)
Comment (ohessel) [reply README.adoc:54]: What I wanted to say is: 
The operator framework can be used to do the ground work, other than implementing the whole workflow/api by oneself.

File: README.adoc (hunk 6 above)
Comment (ohessel) [reply README.adoc:84]: Same as the next one - there's rust, java is discontinued as operator-language (at least for now)

File: README.adoc (hunk 10 above)
Comment (ohessel) [reply README.adoc:97]: 👍

File: README.adoc (hunk 4 above)
Comment (ohessel) [reply README.adoc]: I re-formulated the Agent and also Operator part, so the picture is clearer

**Review by lfrancke — APPROVED** [review]
Thank you!

**Pull request merged**